crates/astria-conductor/src/executor.rs

```diff
@@ -487,4 +507,37 @@ mod test {
         // should be empty because nothing was executed
         assert!(executor.sequencer_hash_to_execution_hash.is_empty());
     }
+
+    #[tokio::test]
+    async fn handle_block_received_from_data_availability_not_yet_executed_no_relevant_transactions_execute_block()
+     {
+        let mut conf = test_config();
+        conf.execute_empty_blocks = true; // empty block will be executed
+        let finalized_blocks = Arc::new(Mutex::new(HashSet::new()));
+        let execution_client = MockExecutionClient {
+            finalized_blocks: finalized_blocks.clone(),
+        };
+        let (mut executor, _) = Executor::new(execution_client, &conf).await.unwrap();
+
+        let block: SequencerBlockSubset = get_test_block_subset();
+        let expected_execution_state = hash(&executor.execution_state);
+
+        executor
+            .handle_block_received_from_data_availability(block)
+            .await
+            .unwrap();
+
+        // should not have executed or finalized the block
+        assert!(!finalized_blocks.lock().await.is_empty());
```

Comment:
I don't think this assertion is necessary. If `finalized_blocks` was empty, the next assertion would also fail.